olive/resource_path.py

```diff
@@ -316,64 +317,126 @@ def save_to_dir(self, dir_path: Union[Path, str], name: str = None, overwrite: b
-            ),
-            "datastore_name": ConfigParam(type_=str, required=True, description="Name of the datastore."),
-            "relative_path": ConfigParam(type_=str, required=True, description="Relative path to the resource."),
+            "azureml_client": ConfigParam(type_=AzureMLClientConfig, description="AzureML client config."),
+            "datastore_name": ConfigParam(type_=str, description="Name of the datastore."),
+            "relative_path": ConfigParam(type_=str, description="Relative path to the resource."),
+            "datastore_url": ConfigParam(type_=str, description="URL of the datastore."),
         }
 
     def get_path(self) -> str:
-        workspace_config = self.config.azureml_client.get_workspace_config()
-        return (
-            f"{_get_azureml_resource_prefix(workspace_config)}"
-            f"/datastores/{self.config.datastore_name}/paths/{self.config.relative_path}"
-        )
+        if not self.config.datastore_url:
+            workspace_config = self.config.azureml_client.get_workspace_config()
+            return (
+                f"{_get_azureml_resource_prefix(workspace_config)}"
+                f"/datastores/{self.config.datastore_name}/paths/{self.config.relative_path}"
+            )
+        return self.config.datastore_url
+
+    def is_file(self, fsspec) -> bool:
+        return fsspec.info(self.get_relative_path()).get("type") == "file"
+
+    def get_relative_path(self) -> str:
+        if not self.config.relative_path:
+            assert self.config.datastore_url
+            self.config.relative_path = re.split("/datastores/.*/paths/", self.config.datastore_url)[-1]
+        return self.config.relative_path
+
+    def get_aml_client_config(self) -> AzureMLClientConfig:
+        if not self.config.azureml_client:
+            subscription_id = re.split("/subscriptions/", self.config.datastore_url)[-1].split("/")[0]
+            resource_group = re.split("/resourcegroups/", self.config.datastore_url)[-1].split("/")[0]
+            workspace_name = re.split("/workspaces/", self.config.datastore_url)[-1].split("/")[0]
+            self.config.azureml_client = AzureMLClientConfig(
+                subscription_id=subscription_id,
+                resource_group=resource_group,
+                workspace_name=workspace_name,
+            )
+        return self.config.azureml_client
 
     def save_to_dir(self, dir_path: Union[Path, str], name: str = None, overwrite: bool = False) -> str:
         # there is no direct way to download a file from a datastore
         # so we will use a workaround to download the file by creating a aml model
         # that references the file and downloading the model
-        from azure.ai.ml.constants import AssetTypes
-        from azure.ai.ml.entities import Model
-        from azure.core.exceptions import ServiceResponseError
+        try:
+            from azureml.fsspec import AzureMachineLearningFileSystem
```

Comment:
need we add an item in the extra?

Reply:
Yes, I think we can add `azureml-fsspec` to the pre-existing `azureml` extra?

Reply:
the azureml-fsspec would introduce azureml-dataset which depends on azureml sdk1.

Reply:
![image](https://github.com/microsoft/Olive/assets/13343117/fe8b173d-e1f5-4f9d-a00f-4dc17fc04fc0)
No dependencies on azureml-core. The dependent package size is about 260Mb.